packages/boot/src/__tests__/acceptance/application-metadata.booter.acceptance.ts

```diff
@@ -33,14 +29,18 @@ describe('application metadata booter acceptance tests', () => {
     // Add the following files
     // - package.json
     // - dist/application.js
+
+    const appJsFile = resolve(__dirname, '../fixtures/application.js');
+    let appJs = fs.readFileSync(appJsFile, 'utf-8');
+    // Adjust the relative path for `import`
+    appJs = appJs.replace('../..', '../../..');
```

Comment:
Can we extract this logic into `sandbox.copyFile` please? IIRC, `copyFile` is already detecting JS files and updating sourcemap URL to accommodate the new location, it would be great if we could do the same for import paths.

If it's too work to detect which paths to update, then perhaps `copyFile` can accept a new `options` to specify the mapping? Example usage:

```ts
await sandbox.copyFile(
  resolve(__dirname, '../fixtures/application.js'),
  'dist/application.js',
  {
    transform: text => text.replace('../..', '../../..'),
  },
});
```

WDYT?

Reply:
Maybe we should defer it. The current `copyFile` does not read the source file and we don't even know if the source is a text or binary file.

Reply:
For context, here is the current `copyFile` implementation, you are right that we are not reading the source file.

https://github.com/strongloop/loopback-next/blob/862072b82ad88ec4b248e0b243dda58b53640f94/packages/testlab/src/test-sandbox.ts#L135-L146

I have two concerns about your current version:

(1)
The code is a bit involved and at different level of abstraction when compared the surrounding lines.

I'd like to read `getApp` as a sequence of instructions for copying files, I don't want to be distracted with implementation details of transforming source file content into target.

```ts
await sandbox.copyFile(resolve(__dirname, '../fixtures/application.js'), {
  // Adjust the relative path for `import`
  transform: text => text.replace('../..', '../../..'),
});
await sandbox.copyFile(resolve(__dirname, '../fixtures/package.json'));
```

(2) 
The code modifying `application.js` is duplicated in two test suites. These two places can easily get out of sync and it's likely additional copies will be introduced in the future as more test files are added.

Can you at least extract the code dealing with `application.js` into a shared test helper, to avoid duplication?